src/clojure/nrepl/middleware/session.clj

```diff
@@ -271,8 +266,12 @@
                         (reset-state)
                         running))))
      :close #(threading/interrupt-stop (:thread @state))
-     :exec (fn [exec-id r ack & [msg]]
-             (.put ^LinkedBlockingQueue (:queue @state) [exec-id r ack msg]))}))
+     :exec (fn [exec-id r ack]
+             ;; Here, *msg* is bound by session middleware on the server/handler
+             ;; thread. We have to convey it to the executor thread.
+             (if *msg*
```

Comment:
Why do check that `*msg*` might be unbound?

Reply:
Defensive programming here. It can't be unbound (I hope) in the current implementation, but if something changes in the future and inadvertently breaks this invariant, at least we'll get a reasonable error message instead of what happens now.

Reply:
Got it.